code/controllers/subsystem/ghost_spawns.dm

```diff
@@ -0,0 +1,216 @@
+SUBSYSTEM_DEF(ghost_spawns)
+	name = "Ghost Spawns"
+	init_order = INIT_ORDER_EVENTS
+	flags = SS_BACKGROUND
+	wait = 1 SECONDS
+	runlevels = RUNLEVEL_GAME
+	offline_implications = "Ghosts will no longer be able to respawn as event mobs (Terror, Blob, etc..). Shuttle call recommended."
+
+	var/list/datum/candidate_poll/currently_polling // List of polls currently ongoing, to be checked on next fire()
+	var/polls_active = FALSE // Whether there are active polls or not
+	var/total_polls = 0 // Number of polls performed since the start
+	var/datum/candidate_poll/next_poll_to_finish // The poll that's closest to finishing
+
+/datum/controller/subsystem/ghost_spawns/fire()
+	if(!polls_active)
+		return
+
+	for(var/poll in currently_polling)
+		var/datum/candidate_poll/P = poll
+		if(P.time_left() <= 0)
+			polling_finished(P)
+
+/datum/controller/subsystem/ghost_spawns/Initialize(start_timeofday)
+	currently_polling = list()
```

Comment:
Define this as a list in the subsystem definition, or make it a lazy list and remove the need for an Initialize() proc